test/integration/get_authorized_entries_integration_test.go

```diff
@@ -0,0 +1,60 @@
+package integration
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/spiffe/spire/pkg/common/pb"
+	"github.com/spiffe/spire/test/integration/setup"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetAuthorizedEntries(t *testing.T) {
+	// Setup SPIRE server and agent
+	srv := setup.NewSpireServer(t)
+	defer srv.Stop()
+
+	agent := setup.NewSpireAgent(t, srv)
+	defer agent.Stop()
+
+	// Create a Join Token
+	joinToken := "test-join-token"
+	_, err := srv.CreateJoinToken(&pb.JoinToken{
+		Token: joinToken,
+		Ttl:   600, // Token validity in seconds
+	})
+	require.NoError(t, err)
+
+	// Attest the agent using the join token
+	err = agent.AttestWithJoinToken(joinToken)
+	require.NoError(t, err)
+
+	// Create a Node-Alias-based registration entry
```

Comment:
maybe I have forgotten what terms mean, but is this actually a node alias-based entry?

```
	parentID := "spiffe://example/spire/agent/tpm/xxxxx"
	spiffeID := "spiffe://example/nodename/foo"
```

this seems like a direct assignment of a registration to a node

a node alias test would be like two registrations, like:

registration to the alias
```
	parentID := "spiffe://example/spire/agent/foo-legal-group"
	spiffeID := "spiffe://example/nodename/foo"
```

alias of agent
```
	parentID := "spiffe://example/spire/server"
	spiffeID := "spiffe://example/spire/agent/foo-legal-group"
        selector := "baseagentid:spiffe://example/spire/agent/tpm/xxxxx"
```

I think both cases are worth testing